api/auth/index.ts

```diff
@@ -1,181 +1,157 @@
-const logger = log4js.getLogger('auth')
-logger.level = config.logging
-const idamUrl = config.services.idamApi
-
-export async function attach(req: EnhancedRequest, res: express.Response, next: express.NextFunction) {
-    const session = req.session!
-    const accessToken = req.cookies[config.cookies.token]
-
-    let expired
-
-    if (accessToken) {
-        const jwtData = jwtDecode(accessToken)
-        const expires = new Date(jwtData.exp).getTime()
-        const now = new Date().getTime() / 1000
-        expired = expires < now
+import axios from 'axios'
+import { NextFunction, Request, Response, Router } from 'express'
+import * as net from 'net'
+import {Client, ClientMetadata, Issuer, Strategy, TokenSet, UserinfoResponse} from 'openid-client'
+import * as passport from 'passport'
+import * as process from 'process'
+import {app} from '../application'
+import {router as keepAlive} from '../keepalive'
+import {config} from '../lib/config'
+import * as log4jui from '../lib/log4jui'
+import {propsExist} from '../lib/objectUtilities'
+import {userHasAppAccess} from './manageCasesUserRoleAuth'
+
+export const router = Router({mergeParams: true})
+
+const cookieToken = config.cookies.token
+const cookieUserId = config.cookies.userId
+const idamURl = config.services.idamApi
+
+const logger = log4jui.getLogger('auth')
+
+export async function configureIssuer(url: string) {
+    let issuer: Issuer<Client>
+
+    logger.info('getting oidc discovery endpoint')
+    issuer = await Issuer.discover(`${url}/o`)
+
+    const metadata = issuer.metadata
+    metadata.issuer = config.services.iss
+
+    return new Issuer(metadata)
+}
+
+export async function configure(req: Request, res: Response, next: NextFunction) {
+
+    const host = req.get('host')
+    const fqdn = req.protocol + '://' + host
+
+    // we don't want to configure strategy if coming from direct IP address (e.g. could be health endpoint)
+    if (net.isIP(host)) {
+        return next()
     }
-    if (!accessToken || expired) {
-        logger.info('Auth Token expired!')
-        doLogout(req, res, 401)
-    } else {
-        const check = await sessionChainCheck(req, res, accessToken)
-        if (check) {
-            logger.info('Attaching auth')
-            // also use these as axios defaults
-            logger.info('Using Idam Token in defaults')
-            axios.defaults.headers.common.Authorization = `Bearer ${session.auth.token}`
-            const token = await asyncReturnOrError(serviceTokenGenerator(), 'Error getting s2s token', res, logger)
-            if (token) {
-                logger.info('Using S2S Token in defaults')
-
-              axios.defaults.headers.common.ServiceAuthorization = token
-            } else {
-                logger.warn('Cannot attach S2S token ')
-                doLogout(req, res, 401)
-            }
-        }
 
-        next()
+    // if client is already configured, exit early
+    if (app.locals.client) {
```

Comment:
I've had to remove this fyi, for some reason it was still configuring the client with an IP address despite the test above